libControls/ToolTip.cpp

```diff
@@ -42,7 +43,21 @@ void ToolTip::add(Control& c, const std::string& str)
 
 void ToolTip::buildDrawParams(std::pair<Control*, std::string>& item, int mouseX)
 {
-	const auto toolTipSize = mFont.size(item.second) + PaddingSize * 2;
+	const auto toolTipLineHeight = mFont.height();
+	const auto numberOfLines = static_cast<int>(std::count(item.second.begin(), item.second.end(), '\n') + 1);
+	const auto toolTipHeight = toolTipLineHeight * numberOfLines + PaddingSize.y * 2;
+
+	auto calculateMaxWidth = [this](const std::string& text) {
+		int maxWidth = 0;
+		processLines(text, [this, &maxWidth](const std::string& line) {
+			maxWidth = std::max(maxWidth, mFont.size(line).x);
+		});
+		return maxWidth;
+	};
+
+	const auto toolTipWidth = calculateMaxWidth(item.second) +PaddingSize.x * 2;
+
+	const auto toolTipSize = NAS2D::Vector<int>{toolTipWidth, toolTipHeight};

```

Comment:
Was just thinking, the individual offsets using `PaddingSize.x * 2` and `PaddingSize.y * 2` could have been handled at the end.

```cpp
const auto toolTipSize = NAS2D::Vector{toolTipWidth, toolTipHeight} + PaddingSize * 2;
```

Though at that point, maybe the intermediate values would be better named something like `textWidth` and `textHeight`.

----

I think the type parameter `<int>` can be inferred when both values are `int`.